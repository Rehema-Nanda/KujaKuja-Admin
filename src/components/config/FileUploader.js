import React from "react";
import reactCSS from "reactcss";
import PropTypes from "prop-types";

import "../Dashboard.css";
import "../../index.css";

const editIcon = require("../../img/config/edit-icon.svg");

class FileUploader extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            url: this.props.url,
            file: this.props.file,
            uploadTitle: this.props.uploadTitle,
        };
    }

    componentDidUpdate(props) {
        const { refresh, url, file } = this.props;

        if (props.refresh !== refresh) {
            this.toggleStates(refresh, url, file);
        }
    }

    toggleStates = (refresh, url, file) => {
        this.setState({
            url: url,
            file: file,
            refresh: refresh,
        });
    }

    onChangeHandler = (event) => {
        const reader = new FileReader();
        const { id, updateFile } = this.props;
        const [files] = event.target.files;

        reader.onload = (e) => {
            const imgElement = document.getElementById(id);
            imgElement.src = e.target.result;
            imgElement.style.display = "inline-block";
        };
        if (files) {
            reader.readAsDataURL(files);
            updateFile(files);
        }
    };

    render() {
        const { url } = this.state;
        const { id, altTag, uploadTitle } = this.props;
        const styles = reactCSS({
            default: {
                title_parent: {
                    textAlign: "center",
                    width: "100%",
                },
                title: {
                    color: "#5b5656",
                    fontSize: "15px",
                    marginTop: "-8px",
                    textAlign: "center",
                },
                file_upload_file: {
                    position: "absolute",
                    top: 0,
                    right: 0,
                    left: 0,
                    bottom: 0,
                    width: "100%",
                    fontSize: "20px",
                    cursor: "pointer",
                    opacity: 0,
                    filter: "alpha(opacity=0)",
                },
                file_upload_image_parent: {
                    position: "relative",
                },
                file_upload_image: {
                    maxWidth: "100px",
                    maxHeight: "100px",
                    minWidth: "52px",
                    minHeigth: "52px",
                    margin: "auto",
                    boxSizing: "none",
                    display: url ? "inline-block" : "none",
                    position: "relative",
                },
                file_upload_parent: {
                    borderRadius: "5px",
                    backgroundColor: "#ebebeb",
                    padding: "10px",
                    minWidth: "300px",
                    position: "relative",
                },
                edit_icon: {
                    textAlign: "center",
                    width: "26px",
                    height: "26px",
                    backgroundColor: "#FFC300",
                    borderRadius: "5px",
                    position: "absolute",
                    bottom: "-8px",
                    right: "-8px",
                },
                edit_icon_image: {
                    width: "12px",
                    margin: "auto",
                },
            },
        });

        return (
            <div style={styles.file_upload_parent} className="d-flex justify-content-start">
                <div>
                    <div style={styles.file_upload_image_parent}>
                        <img
                            alt={altTag}
                            id={id}
                            style={styles.file_upload_image}
                            src={url}
                        />
                        <div style={styles.edit_icon}>
                            <img alt="Edit Icon" style={styles.edit_icon_image} src={editIcon} />
                        </div>
                    </div>
                </div>
                <div style={styles.title_parent}>
                    <h3 style={styles.title}>{uploadTitle}</h3>
                    <input name="file" onChange={this.onChangeHandler} type="file" style={styles.file_upload_file} />
                </div>
            </div>
        );
    }
}

FileUploader.propTypes = {
    url: PropTypes.string.isRequired,
    file: PropTypes.string.isRequired,
    uploadTitle: PropTypes.string.isRequired,
    refresh: PropTypes.bool.isRequired,
    id: PropTypes.string.isRequired,
    updateFile: PropTypes.func.isRequired,
    altTag: PropTypes.string.isRequired,
};

export default FileUploader;
