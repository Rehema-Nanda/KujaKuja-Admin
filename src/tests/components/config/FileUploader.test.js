import React from "react";
import { shallow } from "enzyme";
import FileUploader from "../../../components/config/FileUploader";

describe("<FileUploader />", () => {
    const testProps = {
        updateFile: jest.fn(),
        file: "",
        url: "",
        refresh: "test",
    };
    const wrapper = shallow(<FileUploader {...testProps} />);

    it("renders corectly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should call onChangeHandler function when the input changes", () => {
        const testFunc = jest.spyOn(wrapper.instance(), "onChangeHandler");
        const blob = new Blob();
        const event = {
            target: {
                files: [blob],
            },
        };
        wrapper.instance().forceUpdate(); // reason here https://github.com/airbnb/enzyme/issues/944
        const input = wrapper.find("input");
        input.simulate("change", event);
        expect(testFunc).toHaveBeenCalled();
    });

    it("changes state when the component props changes", () => {
        wrapper.setProps({ refresh: "something" });
        expect(wrapper.state().refresh).toEqual("something");
    });
});
