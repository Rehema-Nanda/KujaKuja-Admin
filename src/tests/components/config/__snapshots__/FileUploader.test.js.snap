// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUploader /> renders corectly 1`] = `
"<div style={{...}} className=\\"d-flex justify-content-start\\">
  <div>
    <div style={{...}}>
      <img alt={[undefined]} id={[undefined]} style={{...}} src=\\"\\" />
      <div style={{...}}>
        <img alt=\\"Edit Icon\\" style={{...}} src={{...}} />
      </div>
    </div>
  </div>
  <div style={{...}}>
    <h3 style={{...}} />
    <input name=\\"file\\" onChange={[Function]} type=\\"file\\" style={{...}} />
  </div>
</div>"
`;
